metadata/cache.go

```diff
@@ -52,23 +51,19 @@ const DefaultEndpointPath = "api/v1/targets/metadata"
 const MetricTypeUntyped = "untyped"
 
 type Entry struct {
-	scrape.MetricMetadata
-	ValueType metric_pb.MetricDescriptor_ValueType
-}
-
-func (e *Entry) Metric() string {
-	return e.MetricMetadata.Metric
-}
-
-func (e *Entry) MetricType() textparse.MetricType {
-	return e.MetricMetadata.Type
+	Metric     string
+	MetricType textparse.MetricType
+	ValueType  metric_pb.MetricDescriptor_ValueType
+	Help       string
 }
 
 // NewEntry returns a new metadata entry.
 func NewEntry(metric string, metricType textparse.MetricType, valueType metric_pb.MetricDescriptor_ValueType, help string) *Entry {
```

Comment:
Optional: Omit this constructor.

Reply:
Done. That was a lot of work, hopefully for the better :)